src/Cli.Tests/EndToEndTests.cs

```diff
@@ -720,4 +721,32 @@ public void TestHelpWriterOutput(string command, string flags, string[] expected
 
         process.Kill();
     }
+
+    /// <summary>
+    /// Test to verify that when base-route is configured, the runtime config is only successfully generated when the
+    /// authentication provider is Static Web Apps.
+    /// </summary>
+    /// <param name="authProvider">Authentication provider specified for the runtime.</param>
+    [DataTestMethod]
+    [DataRow("StaticWebApps")]
+    [DataRow("AppService")]
+    [DataRow("AzureAD")]
```

Comment:
For non -Easy Auth providers, you'll need to supply values for `--auth.issuer` and `--auth.audience` to the CLI otherwise you get the errors:
```
Error: Authentication providers other than EasyAuth and Simulator require both Audience and Issuer.
Error: Could not generate config file.
```

While this test uses the presence of the generated config file to confirm behavior, it doesn't necessarily validate that any failures were because of the base-route configuration value provided which are indicated with your added messages:

```
Error: Runtime base-route can only be specified when authentication provider is Static Web Apps.
Error: Could not generate config file.
```

Reply:
[DataRow("AzureAD")] is still a valid and necessary test for coverage, isn't it? Just need additional logic to handle adding the extra params to the CLI call in that situation.

Reply:
I have this in progress, needed some more refactor for it.

Reply:
done